examples/client/object_server.c

```diff
@@ -224,6 +224,50 @@ static uint8_t prv_get_value(lwm2m_data_t * dataP,
     }
 }
 
+static void remove_optional_resource(uint16_t *resList, int val, int id, int *nbRes) {
+    int i;
+    if (val < 0) {
+        for (i = 0; i < *nbRes; i++) {
+            if (resList[i] == id) {
+                *nbRes -= 1;
+                memmove(&resList[i], &resList[i + 1], (*nbRes - i) * sizeof(resList[i]));
+                break;
+            }
+        }
+    }
+}
+
+static void setup_resources(lwm2m_data_t *const *dataArrayP, const uint16_t *resList, int nbRes) {
+    for (int i = 0; i < nbRes; i++) {
+        (*dataArrayP)[i].id = resList[i];
+    }
+}
+
+static void remove_all_optional_resources(uint16_t *resList, server_instance_t *targetP, int *nbRes) {
```

Comment:
Constification + using a more fitting data type:
```suggestion
static void remove_all_optional_resources(uint16_t *const resList, const server_instance_t *const targetP,
                                          size_t *const nbRes) {
```